File: Python/Entities/eStationConfig.py
import msgpack
from dataclasses import dataclass, field
from typing import List

@dataclass
class eStationConfig:
    """
    eStation configuration
    """
    Alias: str = ""
    Server: str = "192.168.4.92"
    ConnParam: List[str] = field(default_factory=list)
    Encrypt: bool = False
    AutoIP: bool = True
    LocalIP: str = ""
    Subnet: str = ""
    Gateway: str = ""
    Heartbeat: int = 15

    def to_msgpack(self) -> bytes:
        data = [
            self.Alias,
            self.Server,
            self.ConnParam,
            self.Encrypt,
            self.AutoIP,
            self.LocalIP,
            self.Subnet,
            self.Gateway,
            self.Heartbeat
        ]
        return msgpack.packb(data)

    @staticmethod
    def from_msgpack(packed: bytes):
        data = msgpack.unpackb(packed)
        return eStationConfig(
            Alias=data[0],
            Server=data[1],
            ConnParam=data[2],
            Encrypt=data[3],
            AutoIP=data[4],
            LocalIP=data[5],
            Subnet=data[6],
            Gateway=data[7],
            Heartbeat=data[8]
        )

File: Java/src/main/java/com/estation/demo/entities/EStationConfig.java
package com.estation.demo.entities;

import com.estation.demo.util.MessagePackUtil;
import java.util.ArrayList;
import java.util.List;

public record EStationConfig(
        String alias,
        String server,
        List<String> connParam,
        boolean encrypt,
        boolean autoIP,
        String localIP,
        String subnet,
        String gateway,
        int heartbeat
) {
    public byte[] toMessagePack() {
        return MessagePackUtil.pack(List.of(
                alias,
                server,
                connParam,
                encrypt,
                autoIP,
                localIP,
                subnet,
                gateway,
                heartbeat
        ));
    }

    public static EStationConfig fromMessagePack(byte[] data) {
        List<Object> values = MessagePackUtil.asList(MessagePackUtil.unpack(data));
        return new EStationConfig(
                MessagePackUtil.asString(values.get(0)),
                MessagePackUtil.asString(values.get(1)),
                MessagePackUtil.asStringList(values.get(2)),
                MessagePackUtil.asBoolean(values.get(3)),
                MessagePackUtil.asBoolean(values.get(4)),
                MessagePackUtil.asString(values.get(5)),
                MessagePackUtil.asString(values.get(6)),
                MessagePackUtil.asString(values.get(7)),
                MessagePackUtil.asInt(values.get(8))
        );
    }
}
